### PR TITLE
Abort publish if no dist folder

### PR DESCRIPTION
**Pull request by krishnaglick**

### CODE REVIEW

File: bin/publish [hunk 1]
```diff
@@ -124,6 +124,14 @@ rm -rf ui/*/.next
 pnpm install --prefer-frozen-lockfile
 git checkout pnpm-lock.yaml # the other github action will take care of this, but we need a clean working directory to publish
 
+# Check to make sure it was built correctly
+rm -rf core/dist
+if [ ! -d "core/dist" ]; then
```
Comment (bleonard) [inline bin/publish]: think this could fail on other plugins too?

Comment (krishnaglick) [reply bin/publish]: Added a few more checks

File: bin/publish [hunk 2]
```diff
@@ -124,6 +124,22 @@ rm -rf ui/*/.next
 pnpm install --prefer-frozen-lockfile
 git checkout pnpm-lock.yaml # the other github action will take care of this, but we need a clean working directory to publish
 
+# Check to make sure it was built correctly
+rm -rf core/dist
+if [ ! -d "core/dist" ] ||\
+   [ ! -d "cli/dist" ] ||\
+   [ ! -d "ui/ui-community/.next" ] ||\
+   [ ! -d "ui/ui-config/.next" ] ||\
+   [ ! -d "ui/ui-enterprise/.next" ] ||\
```
Comment (evantahler) [inline bin/publish:136]: I like testing the UI packages, core, and cli explicitly

File: bin/publish [hunk 3]
```diff
@@ -124,6 +124,22 @@ rm -rf ui/*/.next
...
+   [ ! -d "plugins/postgres/dist" ] ||\
+   [ ! -d "plugins/mysql/dist" ] ||\
+   [ ! -d "plugins/mongo/dist" ] ||\
+   [ ! -d "plugins/bigquery/dist" ]; then
```
Comment (evantahler) [inline bin/publish]: Rathe then check some of these plugins, you could check them all.  There are lots of ways, but here's one suggestion:

```bash
# Count up the number of plugins with `src` folders:
PLUGINS=`find 'plugins' -name src -and -type d | wc -l`
# Count the number of dist folders in the plugins directory
BUILT_PLUGINS=`find 'plugins' -name dist -and -type d | wc -l`
# Check if they are equal
if [[ "$PLUGINS" != "$BUILT_PLUGINS" ]]; then echo 'OH NO'; fi
```

**Review by evantahler — CHANGES_REQUESTED** [review]

File: bin/publish [hunk 4]
```diff
@@ -124,6 +124,22 @@ rm -rf ui/*/.next
 pnpm install --prefer-frozen-lockfile
 git checkout pnpm-lock.yaml # the other github action will take care of this, but we need a clean working directory to publish
 
+# Check to make sure it was built correctly
+rm -rf core/dist
```
Comment (pedroslopez) [inline bin/publish]:
```suggestion
```

was this left in from a test? looks like this explicitly removes the `dist` dir from core

**Review by pedroslopez — CHANGES_REQUESTED** [review]

File: bin/publish (hunk 4 above)
Comment (krishnaglick) [reply bin/publish]: Nice catch, it was!

File: bin/publish [hunk 5]
```diff
@@ -124,6 +124,22 @@ rm -rf ui/*/.next
 pnpm install --prefer-frozen-lockfile
 git checkout pnpm-lock.yaml # the other github action will take care of this, but we need a clean working directory to publish
 
+# Check to make sure it was built correctly
+
+PLUGINS=`find 'plugins' -name src -and -type d | wc -l`
+BUILT_PLUGINS=`find 'plugins' -name dist -and -type d | wc -l`
```
Comment (krishnaglick) [inline bin/publish:130]: Thanks for the sick bash-foo Evan

**Review by evantahler — APPROVED** [review]
🐧

File: bin/publish [hunk 6]
```diff
@@ -124,6 +124,22 @@ rm -rf ui/*/.next
...
+
+if [ ! -d "core/dist" ] ||\
+   [ ! -d "cli/dist" ] ||\
+   [ ! -d "ui/ui-community/.next" ] ||\
+   [ ! -d "ui/ui-config/.next" ] ||\
+   [ ! -d "ui/ui-enterprise/.next" ] ||\
+   [ "$PLUGINS" != "$BUILT_PLUGINS" ]; then
+  echo "Missing dist folder(s), exiting"
```
Comment (bleonard) [inline bin/publish:138]: might be nice in some future iteration to see which are missing.

**Pull request merged**